packages/strapi-utils/lib/build-query.js

```diff
@@ -103,33 +103,40 @@ const hasDeepFilters = (whereClauses = [], { minDepth = 1 } = {}) => {
 };
 
 const normalizeClauses = (whereClauses, { model }) => {
-  return whereClauses
-    .filter(({ value }) => !_.isNil(value))
-    .map(({ field, operator, value }) => {
-      if (BOOLEAN_OPERATORS.includes(operator)) {
-        return {
-          field,
-          operator,
-          value: value.map(clauses => normalizeClauses(clauses, { model })),
-        };
-      }
-
-      const { model: assocModel, attribute } = getAssociationFromFieldKey({
-        model,
-        field,
-      });
-
-      const { type } = _.get(assocModel, ['allAttributes', attribute], {});
-
-      // cast value or array of values
-      const castedValue = castInput({ type, operator, value });
-
+  return whereClauses.map(({ field, operator, value }) => {
+    if (BOOLEAN_OPERATORS.includes(operator)) {
       return {
-        field: normalizeFieldName({ model, field }),
+        field,
         operator,
-        value: castedValue,
+        value: value.map(clauses => normalizeClauses(clauses, { model })),
```

Comment:
if the value is undefined or null the map will fail

Reply:
@alexandrebodin I moved the `isNil` check before the `map`.

Reply:
It was to maintain that example 4 got the same error message as example 3. (I edited the reply.) Now the undefined value error has priority over the unknown field error.